pkg/objstore/swift/swift.go

```diff
@@ -299,30 +322,34 @@ func NewTestContainer(t testing.TB) (objstore.Bucket, func(), error) {
 				"needs to be manually cleared. This means that it is only useful to run one test in a time. This is due " +
 				"to safety (accidentally pointing prod container for test) as well as swift not being fully strong consistent.")
 		}
-
+		c, err := NewContainerFromConfig(log.NewNopLogger(), config, false)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "swift initializing new container")
+		}
 		if err := c.Iter(context.Background(), "", func(f string) error {
-			return errors.Errorf("container %s is not empty", config.ContainerName)
+			return errors.Errorf("container %s is not empty", c.Name())
 		}); err != nil {
-			return nil, nil, errors.Wrapf(err, "swift check container %s", config.ContainerName)
+			return nil, nil, errors.Wrapf(err, "swift check container %s", c.Name())
 		}
-
-		t.Log("WARNING. Reusing", config.ContainerName, "container for Swift tests. Manual cleanup afterwards is required")
+		t.Log("WARNING. Reusing", c.Name(), "container for Swift tests. Manual cleanup afterwards is required")
 		return c, func() {}, nil
 	}
-
-	tmpContainerName := objstore.CreateTemporaryTestBucketName(t)
-
-	if err := c.createContainer(tmpContainerName); err != nil {
-		return nil, nil, err
+	config.ContainerName = objstore.CreateTemporaryTestBucketName(t)
+	config.SegmentContainerName = config.ContainerName
+	c, err := NewContainerFromConfig(log.NewNopLogger(), config, true)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "swift initializing new container")
 	}
-
-	c.name = tmpContainerName
-	t.Log("created temporary container for swift tests with name", tmpContainerName)
+	t.Log("created temporary container for swift tests with name", c.Name())
 
 	return c, func() {
 		objstore.EmptyBucket(t, context.Background(), c)
-		if err := c.deleteContainer(tmpContainerName); err != nil {
-			t.Logf("deleting container %s failed: %s", tmpContainerName, err)
+		time.Sleep(time.Second)
```

Comment:
What's the reason for this? :P

Reply:
:neutral_face: TBH I do not remember after that time :sweat_smile: 

I'll investigate, maybe there was an issue with the eventual consistency for tests :/

Reply:
Tested it without the timeout and the tests are passing so I'll just drop it :)
Thanks for noticing it.